medical/models/medical_patient.py

```diff
@@ -101,7 +101,7 @@ def _create_vals(self, vals):
         })
         return vals
 
-    @api.model
+    @api.model_cr_context
     def _get_default_image(self, vals):
         res = super(MedicalPatient, self)._get_default_image(vals)
```

Comment:
> when using delegation inheritance, methods are not inherited, only fields

So I don't think that we get any access to `res_partner._get_default_image`.

Instead, in case we have not set yet the image (in medical*), base.res_partner is setting the vals['image']  in the create: https://github.com/odoo/odoo/blob/10.0/odoo/addons/base/res/res_partner.py#L521

I believe that this is why it works today when not setting a default path / image.